### PR TITLE
Fixed up a small annouincement article

### PR DESCRIPTION
**Pull request by ivanleomk**
| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit ce02dab09bb99f31aba5c49d733060d5a5727e55  | 
|--------|--------|

### Summary:
Updated blog post for Gemini and VertexAI SDKs with tool calling and structured output support, including installation instructions and example code.

**Key points**:
- Added blog post for Gemini and VertexAI SDKs tool calling and structured output support.
- Updated `docs/blog/posts/announcing-gemini-tool-calling-support.md`.
- Updated installation instructions for `instructor[gemini]` and `instructor[vertexai]`.
- Provided example code for using Gemini and VertexAI tool calling.
- Recommended using Gemini SDK over VertexAI SDK due to ease of setup and free token quota.
- Corrected mode setting to `instructor.Mode.VERTEXAI_TOOLS` for VertexAI.
- Updated `pyproject.toml` version to `1.4.1`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: docs/blog/posts/announcing-gemini-tool-calling-support.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+---
+draft: False
+date: 2024-09-03
+authors:
+  - ivanleomk
+---
+
+# Announcing Gemini and VertexAI Tool Calling Support
+
+We've excited to announce that Instructor now supports tool calling for both the Gemini SDK and the VertexAI SDk.
```
Comment (jxnl) [inline docs/blog/posts/announcing-gemini-tool-calling-support.md:10]: typo

File: docs/blog/posts/announcing-gemini-tool-calling-support.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+Let's walk through a simple example of how to use these new features
+
+## Installation
+
+To get started, install the latest version of Instructor. Depending on whether you're using Gemini or VertexAI, you should install the following:
+
+=== "Gemini"
+
+    ```bash
+    uv pip install "instructor[gemini]"
```
Comment (jxnl) [inline docs/blog/posts/announcing-gemini-tool-calling-support.md:23]: why uv? its not the standard its just going t confuse average developers

File: docs/blog/posts/announcing-gemini-tool-calling-support.md (hunk 1 above)
Comment (ivanleomk) [reply docs/blog/posts/announcing-gemini-tool-calling-support.md]: Fixed the typo

File: docs/blog/posts/announcing-gemini-tool-calling-support.md (hunk 2 above)
Comment (ivanleomk) [reply docs/blog/posts/announcing-gemini-tool-calling-support.md]: Ok reverted it back to just use pip

File: docs/blog/posts/announcing-gemini-tool-calling-support.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
+---
+draft: False
+date: 2024-09-03
+authors:
+  - ivanleomk
+---
+
+# Announcing Gemini and VertexAI Tool Calling Support
```
Comment (jxnl) [inline docs/blog/posts/announcing-gemini-tool-calling-support.md:8]:
```suggestion
# Instructor Now Supports Structured Outputs with Gemini and Vertex AI
```

File: docs/blog/posts/announcing-gemini-tool-calling-support.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+We're excited to announce that `instructor` now supports tool calling for both the Gemini SDK and the VertexAI SDK.
```
Comment (jxnl) [inline docs/blog/posts/announcing-gemini-tool-calling-support.md:10]:
```suggestion
We're excited to announce that `instructor` now supports structured outputs using tool calling for both the Gemini SDK and the VertexAI SDK.
```

File: docs/blog/posts/announcing-gemini-tool-calling-support.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+class User(BaseModel):
+    name: str
+    age: int
+
+
+client = instructor.from_gemini(
+    client=genai.GenerativeModel(
+        model_name="models/gemini-1.5-flash-latest", # (1)!
+    ),
+    mode=instructor.Mode.GEMINI_TOOLS, # (2)!
```
Comment (jxnl) [inline docs/blog/posts/announcing-gemini-tool-calling-support.md:55]: this should be default and hidden

File: docs/blog/posts/announcing-gemini-tool-calling-support.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+=== "VertexAI"
+
+    ```bash
+    pip install "instructor[vertexai]"
+    ```
+
+This ensures that you have the necessary dependencies to use the Gemini or VertexAI SDKs with instructor.
+
+## Getting Started
```
Comment (jxnl) [inline docs/blog/posts/announcing-gemini-tool-calling-support.md:34]: make a recommendation on which one to use

Comment (ivanleomk) [reply docs/blog/posts/announcing-gemini-tool-calling-support.md:39]: Recommendation made

We recommend using the Gemini SDK over the VertexAI SDK for two main reasons.

1. Compared to the VertexAI SDK, the Gemini SDK comes with a free daily quota of 1.5 billion tokens to use for developers.
2. The Gemini SDK is significantly easier to setup, all you need is a `GOOGLE_API_KEY` that you can generate in your GCP console. THe VertexAI SDK on the other hand requires a credentials.json file or an OAuth integration to use.

**Review by jxnl — APPROVED** [review]

**Pull request merged**